src/main/java/org/openrewrite/java/testing/junit5/MockitoJUnitToMockitoExtension.java

```diff
@@ -0,0 +1,206 @@
+                                    .build(),
+                            cd.getCoordinates()
+                                    .addAnnotation(Comparator.comparing(J.Annotation::getSimpleName))
+                    );
+
+                    maybeAddImport(extendWithFqn);
+                    maybeAddImport(mockitoExtensionFqn);
+                }
+            }
+
+            return cd;
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            if (method.getType() == null) {
+                return method;
+            }
+
+            if (TypeUtils.isOfClassType(method.getType().getDeclaringType(), MOCKITO_RULE_FQN)) {
+                getCursor().putMessageOnFirstEnclosing(J.MethodDeclaration.class, MOCKITO_RULE_INVOCATION_KEY, method);
+            } else if (TypeUtils.isOfClassType(method.getType().getDeclaringType(), MOCKITO_TEST_RULE_FQN)) {
+                getCursor().putMessageOnFirstEnclosing(J.MethodDeclaration.class, MOCKITO_TEST_RULE_INVOCATION_KEY, method);
+            }
+
+            return method;
+        }
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration methodDecl, ExecutionContext ctx) {
+            J.MethodDeclaration m = super.visitMethodDeclaration(methodDecl, ctx);
+
+            final J.MethodInvocation mockitoRuleInvocation = getCursor().pollMessage(MOCKITO_RULE_INVOCATION_KEY);
+            final J.MethodInvocation mockitoTestRuleInvocation = getCursor().pollMessage(MOCKITO_TEST_RULE_INVOCATION_KEY);
+            if (mockitoRuleInvocation == null && mockitoTestRuleInvocation == null) {
+                return m;
+            }
+
+            if (m.getBody() == null) {
+                return m;
+            }
+
+            final String printedStatementsParam = getPrintedStatements(m.getBody());
+            m = m.withBody(
+                    m.getBody().withTemplate(
+                            template("{#{}}")
+                                    .javaParser(JAVA_PARSER.get())
+                                    .build(),
+                            m.getBody().getCoordinates().replace(),
+                            printedStatementsParam
+                    )
+            );
+
+            m = m.withBody((J.Block) new AutoFormatVisitor<ExecutionContext>().visit(m.getBody(), ctx, getCursor()));
+
+            return m;
+        }
+
+        private String getPrintedStatements(J.Block body) {
+            List<Statement> statements = body.getStatements().stream()
+                    .filter(it -> !isTargetMethodInvocation(it))
+                    .collect(Collectors.toList());
+
+            StringBuilder printedStatements = new StringBuilder();
+            for (Statement stmt : statements) {
+                printedStatements.append(stmt.print()).append(';');
+            }
+
+            return printedStatements.toString();
+        }
```

Comment:
Printing the Statements to a String only to re-parse back into Statements is unnecessary overhead. There's no need to serialize and re-parse when you already have the data in the right format.

You should do something like `m = m.withBody(m.getBody().withStatements(filteredListOfStatements));`.

Reply:
Oh, that's interesting. I'll refactor the code a bit, thanks for letting me know